tests/zfs-tests/tests/functional/cli_root/zfs_set/mountpoint_003_pos.ksh

```diff
@@ -90,26 +90,39 @@ typeset i=0
 while ((i < ${#args[@]})); do
 	if is_linux; then
 		log_must $MOUNT -t zfs -o ${args[$i]} $testfs $tmpmnt
+		
+		msg=$($MOUNT | $GREP "$tmpmnt ")
+		
+		$ECHO $msg | $GREP "${args[((i))]}" > /dev/null 2>&1
+		if (($? != 0)) ; then
+				$ECHO $msg | $GREP "${args[((i-1))]}" > /dev/null 2>&1
```

Comment:
@behlendorf  The specific options are coupled, and some specific options can not show in the label, like `dev`. So we can not make sure the option is right by check the label. Here we can check by the opposite option which can show in the label. In the table above, the first option in every row can show in the label, If we mount with `nodev`, `nodev` will appear in the label, but `dev` can not. So when mount with option `dev`, the first `$ECHO $msg` must not equal 0, then we can check `nodev` if in the label by the second `$ECHO $msg`. If `nodev` in the label, the script is fail, otherwise the filesystem mount with the specific option is success.